src/imitation/algorithms/preference_comparisons.py

```diff
@@ -906,6 +908,158 @@ def _reward_sums(self, fragment_pairs) -> Tuple[np.ndarray, np.ndarray]:
         return np.array(rews1, dtype=np.float32), np.array(rews2, dtype=np.float32)
 
 
+class SynchronousCLIGatherer(PreferenceGatherer):
+    """Queries for human preferences using the command line interface."""
+
+    def __init__(
+        self,
+        video_dir: pathlib.Path,
+        custom_logger: Optional[imit_logger.HierarchicalLogger] = None,
+    ) -> None:
+        """Initialize the human preference gatherer.
+
+        Args:
+            video_dir: directory where videos of the trajectories are saved.
+            custom_logger: Where to log to; if None (default), creates a new logger.
+        """
+        super().__init__(custom_logger=custom_logger)
+        self.video_dir = video_dir
+
+    def __call__(self, fragment_pairs: Sequence[TrajectoryWithRewPair]) -> np.ndarray:
+        """Displays each pair of fragments and asks for a preference.
+
+        It iteratively requests user feedback for each pair of fragments. If in the
+        command line, it will pop out a video player for each fragment. If in a
+        notebook, it will display the videos. Either way, it will request 1 or 2 to
+        indicate which is preferred.
+
+        Args:
+            fragment_pairs: sequence of pairs of trajectory fragments
+
+        Returns:
+            A numpy array of 1 if fragment 1 is preferred and 0 otherwise, with shape
+            (b, ), where b is the length of the input
+        """
+
+        preferences = np.zeros(len(fragment_pairs), dtype=np.float32)
+        for i, (frag1, frag2) in enumerate(fragment_pairs):
```

Comment:
Could also write as something like (not tested, may need to be modified):
```
preferences = [self._display_videos(**frag) for frag in fragment_pairs] 
preferences = np.astype(preferences, dtype=np.float32)
```

Not sure that's actually any better though -- more succinct but also a bit more magical. WDYT Tom & @levmckinney?